api/types/role.go

```diff
@@ -479,14 +479,48 @@ func (r *RoleV6) convertDenyKubernetesResourcesBetweenRoleVersions(resources []K
 		return resources
 	default:
 		v7resources := slices.Clone(resources)
+		var extraResources []KubernetesResource
 		for i, r := range v7resources {
+			// "namespace" kind used to mean "namespaces" and all resources in the namespace.
+			// It is now represented by 'namespaces' for the resource itself and wildcard for
+			// all resources in the namespace.
+			if r.Kind == KindKubeNamespace {
+				r.Kind = Wildcard
+				r.Namespace = r.Name
+				r.Name = Wildcard
+				r.APIGroup = Wildcard
+				v7resources[i] = r
+				extraResources = append(extraResources, KubernetesResource{
+					Kind:  "namespaces",
+					Name:  r.Namespace,
+					Verbs: r.Verbs,
+				})
+				continue
+			}
+			// The namespace field was ignored in v7 for global resources.
+			if r.Namespace != "" && slices.Contains(V7KubernetesClusterWideResourceKinds, r.Kind) {
+				r.Namespace = ""
+			}
 			if k, ok := KubernetesResourcesKindsPlurals[r.Kind]; ok { // Can be empty if the kind is a wildcard.
 				r.Kind = k
 			}
 			r.APIGroup = Wildcard
 			v7resources[i] = r
+			if r.Kind == Wildcard { // If we have a wildcard, inject the clusterwide resources.
```

Comment:
the kind wildcard will be present anw so we will deny everything and then we append extra entries

Reply:
This is because of the case where you have `kind: '*', namespace: 'foo'`. It used to match everything so we maintain the behavior.